packages/shared/src/features/onboarding/steps/FunnelOrganicSignup.tsx

```diff
@@ -58,25 +71,32 @@ export const FunnelOrganicSignup = withIsActiveGuard(
     const isSocialSignupActive =
       isAuthReady &&
       isLoggedIn &&
-      'infoConfirmed' in user &&
-      !user.infoConfirmed &&
-      user?.providers?.some((prov) => prov !== 'password') &&
-      !isEmailSignupActive;
+      !isEmailSignupActive &&
+      isSocialSignupUser(user);
+
     const onAuthStateUpdate = useCallback(
       (data: Partial<AuthProps>) => {
         const { defaultDisplay, isLoginFlow } = data;
+        // capture the default display from the auth state
         if (defaultDisplay) {
           setAuthDisplay(defaultDisplay);
         }
 
+        // Move outside the funnel if is login flow
         if (isLoginFlow) {
           setAuth((prev) => ({
             ...prev,
             isLoginFlow: true,
             isAuthenticating: true,
             defaultDisplay: AuthDisplay.Default,
           }));
+          return;
```

Comment:
when does this case happen in regards to users coming from funnel?

Reply:
Only when clicking `Already have an account? Log in`. We need basically to quit the funnel on this click. 
<img width="500" alt="image" src="https://github.com/user-attachments/assets/1c734bc1-0473-4d43-ab1f-1048e167b598" />